### PR TITLE
[AV] Correct Playback interface so that it contains all common method…

### PR DESCRIPTION
**Pull request by mczernek**
# Why

Documentation includes method for Video reference, which is not available since at least sdk32. Some users reports this inconsistence.

# How

Updating documentation and updating Playback interface for Audio and Video objects.

### CODE REVIEW

File: docs/pages/versions/v35.0.0/sdk/av.md [hunk 1]
```diff
@@ -351,11 +355,11 @@ _onPlaybackStatusUpdate = playbackStatus => {
 };
 
 ... // Load the playbackObject and obtain the reference.
-playbackObject.setOnPlaybackStatusUpdate(this._onPlaybackStatusUpdate);
+sound.setOnPlaybackStatusUpdate(this._onPlaybackStatusUpdate);
 ...
 ```
 
-#### Example: Loop media exactly 20 times
+#### Example: Loop media exactly 20 times for Sound
```
Comment (sjchmiela) [inline docs/pages/versions/v35.0.0/sdk/av.md]:
```suggestion
#### Example: Loop sound exactly 20 times
```

File: docs/pages/versions/v35.0.0/sdk/av.md [hunk 2]
```diff
@@ -351,11 +355,11 @@ _onPlaybackStatusUpdate = playbackStatus => {
 };
 
 ... // Load the playbackObject and obtain the reference.
```
Comment (sjchmiela) [inline docs/pages/versions/v35.0.0/sdk/av.md]:
```suggestion
... // Load the sound and obtain the reference.
```

**Review by sjchmiela — COMMENTED** [review]
On the other hand, maybe we should just expose a `setOnPlaybackUpdateStatus` method from `Video.tsx` (which shouldn't be too difficult, we would just save the callback as an instance variable and call in `_handleNewStatus`?

File: docs/pages/versions/unversioned/sdk/av.md [hunk 3]
```diff
@@ -64,7 +64,7 @@ render() {
 
 ## Playback API
 
-On the `playbackObject` reference, the following API is provided:
+On the `playbackObject` reference, the following API is provided for both Audio and Video:
```
Comment (brentvatne) [inline docs/pages/versions/unversioned/sdk/av.md]:
```suggestion
On the `playbackObject` reference, the following API is provided for both `Audio` and `Video`:
```

File: packages/expo-av/src/AV.ts [hunk 4]
```diff
@@ -250,14 +254,14 @@ export interface Playback extends AV {
  */
 export const PlaybackMixin = {
   async playAsync(): Promise<PlaybackStatus> {
-    return ((this as any) as AV).setStatusAsync({ shouldPlay: true });
+    return ((this as any) as Playback).setStatusAsync({ shouldPlay: true });
```
[the remark this answers is not in the document]
Comment (mczernek) [reply packages/expo-av/src/AV.ts]: My idea was, that we really require `this` to implement `Playback` interface, and `PlaybackMixin` is designed to add implementation to some methods.
So it is sufficient, but would be logically wrong to pass here `AV` implementation. I know that this does not guard us anyhow, but I thought that this way our intentions are expressed more clearly.

File: packages/expo-av/src/Video.tsx [hunk 5]
```diff
@@ -314,6 +314,7 @@ export default class Video extends React.Component<VideoProps, VideoState> imple
   setProgressUpdateIntervalAsync!: (
     progressUpdateIntervalMillis: number
   ) => Promise<PlaybackStatus>;
+  setOnPlaybackUpdateStatus!: (progressUpdateIntervalMillis: number) => Promise<PlaybackStatus>;
```
[the remark this answers is not in the document]
Comment (mczernek) [reply packages/expo-av/src/Video.tsx]: No, is should not. Will delete.

**Comment by mczernek** [comment]
> On the other hand, maybe we should just expose a `setOnPlaybackUpdateStatus` method from `Video.tsx` (which shouldn't be too difficult, we would just save the callback as an instance variable and call in `_handleNewStatus`?

I guess we could do that, but I do not think we want to do it backward. What about just updating docs for now, and creating another issue for adding this method?

**Review by sjchmiela — APPROVED** [review]
Thanks! 👏

**Comment by rusakovic** [comment]
@mczernek 
Helo Michał!
I tried to solve video issues several months. Maybe you can suggest correct manual for AV. 
How to `loadAsync`, `unloadAsync`, and pause/play correctly with `useEffect`, `useState`, `useRef` hooks. Because currently official manual is super outdated.
I'm ready to help!
Thank you!

**Pull request merged**